backend/neofs/neofs.go

```diff
@@ -0,0 +1,804 @@
+package neofs
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	"github.com/nspcc-dev/neofs-sdk-go/container"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	"github.com/nspcc-dev/neofs-sdk-go/policy"
+	"github.com/nspcc-dev/neofs-sdk-go/pool"
+	"github.com/nspcc-dev/neofs-sdk-go/resolver"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/hash"
+	"github.com/rclone/rclone/fs/walk"
+	"github.com/rclone/rclone/lib/bucket"
+)
+
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "neofs",
+		Description: "Distributed, decentralized object storage NeoFS",
+		NewFs:       NewFs,
+		Options: []fs.Option{
+			{
+				Name:     "neofs_endpoint",
+				Help:     "Endpoint to connect to NeoFS node",
+				Required: true,
+			},
+			{
+				Name:    "neofs_connection_timeout",
+				Default: 4 * time.Second,
+				Help:    "NeoFS connection timeout",
+			},
+			{
+				Name:    "neofs_request_timeout",
+				Default: 4 * time.Second,
+				Help:    "NeoFS request timeout",
+			},
+			{
+				Name:    "neofs_rebalance_interval",
+				Default: 15 * time.Second,
+				Help:    "NeoFS rebalance connections interval",
+			},
+			{
+				Name:    "neofs_session_expiration",
+				Default: math.MaxUint32,
+				Help:    "NeoFS session expiration epoch",
+			},
+			{
+				Name:     "rpc_endpoint",
+				Help:     "Endpoint to connect to Neo rpc node",
+				Required: true,
```

Comment:
Can we make NNS resolver optional together with `rpc_endpoint`?

Reply:
How should we treat the named containers then?

Reply:
Like s3-gw: lookup attributes of containers created by the wallet or throw an error.